src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -233,6 +244,29 @@ public partial class Activity
             }
         }
 
+        /// <summary>
+        /// Links is the list of all <see cref="ActivityLink" /> objects which contain the <see cref="ActivityContext" /> attached to this Activity object.
+        /// If there is no any <see cref="ActivityLink" /> object attached to the Activity object, Links will return empty list.
+        /// </summary>
+        public IEnumerable<ActivityLink> Links
+        {
+            get
+            {
+                KeyValueListNode<ActivityLink>? links = _links;
+                return links != null ? Iterate(links) : s_emptyLinks;
+
+                static IEnumerable<ActivityLink> Iterate(KeyValueListNode<ActivityLink>? links)
+                {
+                    do
+                    {
+                        yield return links!.keyValue;
+                        links = links.Next;
+                    }
+                    while (links != null);
```

Comment:
I think I saw an API for manipulating/adding/etc links. What happens if those APIs are used should enumerating?

Reply:
We snap the first linked list item before enumerating. 

      KeyValueListNode<ActivityLink>? links = _links;

If the AddLink API method called after that, the newly added item wouldn't show up in the returned list. 

---
In reply to: [383014200](https://github.com/dotnet/runtime/pull/32660#discussion_r383014200) [](ancestors = 383014200)